plugins/web/opentelemetry-instrumentation-user-interaction/src/types.ts

```diff
@@ -14,7 +14,28 @@
  * limitations under the License.
  */
 
-import * as types from '@opentelemetry/api';
+import { HrTime, Span } from '@opentelemetry/api';
+import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+
+export interface UserInteractionInstrumentationConfig
+  extends InstrumentationConfig {
+  /**
+   * List of events to instrument (like 'mousedown', 'touchend', 'play' etc).
+   * By default only 'click' event is instrumented.
+   */
+  eventTypes?: string[];
+
+  /**
+   * Callback function called each time new span is being created.
+   * You can use this handler to enhance created span with extra attributes.
+   * Return `false` to prevent span recording.
+   */
+  onSpan?: (
```

Comment:
onSpan sounds more like an callback event approach which should return void type. I would name it like `shouldPreventSpanCreation` this way it sounds more intuitive and you can avoid adding `?` everywhere in a code by refactoring the piece in constructor - see my comment there

Reply:
The main purpose of this PR was to add ability to enhance spans and IMO using `shouldPreventSpanCreation` only to add some span attributes may looks unintuitive. On the other hand I agree that `onSpan` should not expect a boolean to be returned. I was inspired by a DOM's event listeners but the 'return false' thing to prevent default behaviour is not a good pattern.
What do you think about leaving `onSpan` with the initial version (without a boolean as a return value) AND adding `shouldPreventSpanCreation` in the current form?

Reply:
@obecny any thoughts on that? After some consideration, single `shouldPreventSpanCreation` will also work for me.

Reply:
I'm just trying to think of different edge cases and having `onSpan` and then `shouldPreventSpanCreation` would confuse user more than it would help. I still think that having just one function might be easier for user and also for us to keep the logic simple. I just suggested to change a name a bit so that it is easier to understand and it will have correct default behaviour (reversing the logic) if you don't specify anything

Reply:
I thought that `shouldPreventSpanCreation` may not suggest that this can be also used to enhance spans, but now I think it will be fine. Thanks :)